processor/src/main/java/org/mapstruct/ap/internal/model/common/Parameter.java

```diff
@@ -34,18 +36,20 @@
     private final Type type;
     private final boolean mappingTarget;
     private final boolean targetType;
+    private final boolean mappingContext;
 
-    public Parameter(String name, Type type, boolean mappingTarget, boolean targetType) {
+    public Parameter(String name, Type type, boolean mappingTarget, boolean targetType, boolean mappingContext) {
```

Comment:
Hmm, sub types just for changing the result of three boolean accessors seems a bit over-dosed to me. How about making this constructor private and provide a more slim constructor or factory method? There's just one caller for this anyway...